libraries/shared/executor/executor.go

```diff
@@ -0,0 +1,154 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package executor
+
+import (
+	"fmt"
+	"plugin"
+	syn "sync"
+	"time"
+
+	"github.com/vulcanize/vulcanizedb/libraries/shared/constants"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/fetcher"
+	storageUtils "github.com/vulcanize/vulcanizedb/libraries/shared/storage/utils"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/transformer"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/watcher"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/fs"
+)
+
+type Executor struct {
+	BlockChain              core.BlockChain
+	DB                      *postgres.DB
+	Plugin                  PluginLookUpper
+	RecheckHeadersArg       bool
+	QueueRecheckInterval    time.Duration
+	PollingInterval         time.Duration
+	EthEventInitializers    []transformer.EventTransformerInitializer
+	EthStorageInitializers  []transformer.StorageTransformerInitializer
+	EthContractInitializers []transformer.ContractTransformerInitializer
+	FileTailer              fs.Tailer
+}
+
+func NewExecutor(db *postgres.DB, blockChain core.BlockChain, plug PluginLookUpper, recheckHeadersArg bool, pollingInterval time.Duration, queueRecheckInterval time.Duration, fileTailer fs.Tailer) Executor {
```

Comment:
not happy about this long list of args - I thought about combining a few into a struct, but that also seemed annoying 🤷‍♀